xskillscore/core/probabilistic.py

```diff
@@ -475,6 +478,81 @@ def threshold_brier_score(
         return res.mean(dim, keep_attrs=keep_attrs)
 
 
+def _check_bin_dim(ds):
+    """Assert that bin_dim is in ds. Try to guess and rename edges dimension."""
+    bin_dim = "category_edge"
+    if bin_dim not in ds.dims:
+        raise ValueError(f"require {bin_dim} dimension, found {ds.dims}")
+    return ds
+
+
+def _check_data_within_edges(forecasts, forecasts_edges):
+    """Check that forecasts_edges has category_edge dimension and forecasts_edges span range of forecasts (per variable if xr.Dataset)."""
+    if "category_edge" not in forecasts_edges.dims:
+        raise ValueError(
+            f"Expect to find category_edge in forecast_edges, found {forecasts_edges.dims}"
+        )
+
+    def _check(forecasts, forecasts_edges):
+        if forecasts.min() < forecasts_edges.min():
+            raise ValueError(
+                f"found forecasts outside forecast_edges, found forecasts_edges.min() = \n{forecasts_edges.min()}\n and forecasts.min() = \n {forecasts.min()}"
+            )
+        if forecasts.max() > forecasts_edges.max():
+            raise ValueError(
+                f"found forecasts outside forecast_edges, found forecasts_edges.max() = \n{forecasts_edges.max()}\n and forecasts.max() = \n {forecasts.max()}"
+            )
+
+    if isinstance(forecasts, xr.Dataset):
+        for v in forecasts.data_vars:
+            _check(forecasts[v], forecasts_edges[v])
+    elif isinstance(forecasts, xr.DataArray):
+        _check(forecasts, forecasts_edges)
+    else:
+        raise ValueError("only defined for xr.DataArrays and xr.Datasets")
+
+
+from .contingency import _get_category_bounds
```

Comment:
shouldn't this go at the top? Would have thought `isort` would have moved it to the top.

Reply:
would've thought the same